common/dynamicconfig/client.go

```diff
@@ -51,6 +51,19 @@ type (
 		GetValue(key Key) []ConstrainedValue
 	}
 
+	// NotifyingClient is an optional interface that a Client can also implement, that adds
+	// support for faster notifications of dynamic config changes.
+	NotifyingClient interface {
+		// Adds a subscription to all updates from this Client. `update` will be called on any
+		// change to the current value set. The caller should call `cancel` to cancel the
+		// subscription. Calls to `update` will not be made concurrently.
+		Subscribe(update ClientUpdateFunc) (cancel func())
```

Comment:
Yeah, I misread your comment at first. I could change the name but it does basically make sense since the subscribe/callback is going in the same direction in both, they're just different layers.